pkg/virt-controller/network/ipamclaims.go

```diff
@@ -0,0 +1,207 @@
+			context.Background(),
+			claim,
+			v1.CreateOptions{},
+		)
+
+		if err != nil {
+			if !k8serrors.IsAlreadyExists(err) {
+				return fmt.Errorf("failed to create IPAMClaim: %w", err)
+			}
+
+			err = m.ensureValidIPAMClaimForVMI(namespace, claim.Name, claim.OwnerReferences[0].UID)
+			if err != nil {
+				return fmt.Errorf("failed validating IPAMClaim: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *IPAMClaimsManager) ensureValidIPAMClaimForVMI(namespace string, claimName string, expectedOwnerUID types.UID) error {
+	currentClaim, err := m.client.IPAMClaimsClient().K8sV1alpha1().IPAMClaims(namespace).Get(
+		context.Background(),
+		claimName,
+		v1.GetOptions{},
+	)
+	if err != nil {
+		return fmt.Errorf("failed getting IPAMClaim: %w", err)
+	}
+
+	if len(currentClaim.OwnerReferences) != 1 || currentClaim.OwnerReferences[0].UID != expectedOwnerUID {
+		return fmt.Errorf("failed validating IPAMClaim, wrong IPAMClaim with the same name still exists")
+	}
+
+	return nil
+}
+
+func composeIPAMClaim(namespace string, ownerRef v1.OwnerReference, ipamClaimParams IPAMClaimParams, interfaceName string) *ipamclaims.IPAMClaim {
+	return &ipamclaims.IPAMClaim{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      ipamClaimParams.ClaimName,
+			Namespace: namespace,
+			OwnerReferences: []v1.OwnerReference{
+				ownerRef,
+			},
+		},
+		Spec: ipamclaims.IPAMClaimSpec{
+			Network:   ipamClaimParams.NetworkName,
+			Interface: interfaceName,
+		},
+	}
+}
+
+func (m *IPAMClaimsManager) GetNetworkToIPAMClaimParams(namespace string, vmiName string, networks []virtv1.Network) (map[string]IPAMClaimParams, error) {
+	nonMultusDefaultNetworks := vmispec.FilterMultusNonDefaultNetworks(networks)
+	nads, err := GetNetworkAttachmentDefinitions(m.client, namespace, nonMultusDefaultNetworks)
+	if err != nil {
+		return nil, fmt.Errorf("failed retrieving network attachment definitions: %w", err)
+	}
+
+	networkToIPAMClaimParams, err := ExtractNetworkToIPAMClaimParams(nads, vmiName)
+	if err != nil {
+		return nil, fmt.Errorf("failed extracting ipam claim params: %w", err)
+	}
+
+	return networkToIPAMClaimParams, nil
+}
+
+func ExtractNetworkToIPAMClaimParams(nadMap map[string]*networkv1.NetworkAttachmentDefinition, vmiName string) (map[string]IPAMClaimParams, error) {
+	networkToIPAMClaimParams := map[string]IPAMClaimParams{}
+	for networkName, nad := range nadMap {
+		persistentIPsNetworkName, err := getPersistentIPsNetworkName(nad)
+		if err != nil {
+			return nil, fmt.Errorf("failed retrieving persistentIPsNetworkName: %w", err)
+		}
+		if persistentIPsNetworkName != "" {
+			networkToIPAMClaimParams[networkName] = IPAMClaimParams{
+				ClaimName:   fmt.Sprintf("%s.%s", vmiName, networkName),
+				NetworkName: persistentIPsNetworkName,
+			}
+		}
+	}
+	return networkToIPAMClaimParams, nil
+}
+
+func getPersistentIPsNetworkName(nad *networkv1.NetworkAttachmentDefinition) (string, error) {
+	if nad.Spec.Config == "" {
+		return "", nil
```

Comment:
Isn't this an error ?

Reply:
well it is true that spec.Config shouldn't be empty, question is if kubevirt should care about it
or just treat it as the knob is disabled
what is not allowed is to unmarsh empty string